extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/OwHandlerFactory.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.binding.onewire.internal;
+
+import static org.eclipse.smarthome.binding.onewire.internal.OwBindingConstants.SUPPORTED_THING_TYPES;
+
+import java.util.HashMap;
+import java.util.Hashtable;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.binding.onewire.internal.discovery.OwDiscoveryService;
+import org.eclipse.smarthome.binding.onewire.internal.handler.DigitalIOThingHandler;
+import org.eclipse.smarthome.binding.onewire.internal.handler.IButtonThingHandler;
+import org.eclipse.smarthome.binding.onewire.internal.handler.MultisensorThingHandler;
+import org.eclipse.smarthome.binding.onewire.internal.handler.OwserverBridgeHandler;
+import org.eclipse.smarthome.binding.onewire.internal.handler.TemperatureSensorThingHandler;
+import org.eclipse.smarthome.config.discovery.DiscoveryService;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link OwHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+@Component(service = ThingHandlerFactory.class, immediate = true, name = "binding.onewire", configurationPolicy = ConfigurationPolicy.OPTIONAL)
```

Comment:
AFAIK the handler factories do not have to be active immediate. See #5511 for reference. Also please add the `configurationPid` property.

Reply:
done